modules/hpe_nimble_acr.py

```diff
@@ -0,0 +1,297 @@
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+# OF ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+# author alok ranjan (alok.ranjan2@hpe.com)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = r'''
+---
+author:
+    - Alok Ranjan (@ranjanal)
+description: "On HPE Nimble Storage Array - Create or delete ACL for volume"
+module: hpe_nimble_acr
+
+options:
+    state:
+        required: True
+        choices:
+            - present
+            - absent
+            - create
+        type: 'str'
+        description:
+        - Choice for ACR operation
+    apply_to:
+        required: False
+        choices:
+            - volume
+            - snapshot
+            - both
+            - pe
+            - vvol_volume
+            - vvol_snapshot
+        type: str
+        default: both
+        description:
+        - Type of object this access control record applies to.
+    chap_user:
+        required: False
+        type: str
+        description:
+        - Name for the CHAP user.
+    lun:
+        required: False
+        type: int
+        description:
+        - If this access control record applies to a regular volume, this attribute is the volume's LUN (Logical Unit Number).
+          If the access protocol is iSCSI, the LUN will be 0. However, if the access protocol is Fibre Channel, the LUN will be in the range from 0 to 2047
+    volume:
+        required: False
+        type: str
+        description:
+        - Name for the volume this access control record applies to.
+    pe_ids:
+        required: False
+        type: list
+        description:
+        - List of candidate protocol endpoints that may be used to access the Virtual Volume. One of them will be selected for the access control record.
+          This field is required only when creating an access control record for a Virtual Volume.
+    protocol_endpoint:
+        required: False
+        type: str
+        description:
+        - Name for the protocol endpoint this access control record applies to.
+    snapshot:
+        required: False
+        type: str
+        description:
+        - Name of the snapshot this access control record applies to.
+    initiator_group:
+        required: False
+        type: str
+        description:
+        - The Initiator Group name.
+
+extends_documentation_fragment: hpe_nimble
+short_description: "Manages a HPE Nimble Storage ACR"
+version_added: "2.9"
+'''
+
+EXAMPLES = r'''
+
+    # if state is create, create ACL for given volume, fails if it exist or cannot create
+    # if state is present, create ACL if not present, else success
```

Comment:
# If state is "create", create ACR for given volume, fails if it exist.
    # If state is "present", create ACR if not already present.

Reply:
ok. changed